File: java/schedule-time-event/src/main/java/com/gearguru/schedule_time_event/Application/Services/BarberShopService.java
package com.gearguru.schedule_time_event.Application.Services;

import com.gearguru.schedule_time_event.Application.Interfaces.IBarberShopService;
import com.gearguru.schedule_time_event.Domain.Entities.Schedule;
import com.gearguru.schedule_time_event.Infrastructure.Interfaces.IBarberShopRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class BarberShopService implements IBarberShopService {
    private final IBarberShopRepository barberShopRepository;

    @Autowired
    public BarberShopService(IBarberShopRepository barberShopRepository) {
        this.barberShopRepository = barberShopRepository;
    }


    @Override
    public Schedule checkIfTimeIsAvailable() {
        ;
        return null;
    }
}

File: csharp/ScheduleTime.Application/Services/BarberShopService.cs
using ScheduleTime.Application.Interfaces;
using ScheduleTime.Application.Services.Contracts;
using ScheduleTime.Domain.Entities;

namespace ScheduleTime.Application.Services
{
    public class BarberShopService : IBaseService<BarberShop>, IBarberShopService
    {
        public Task<BarberShop> VerifyAvailability(BarberShop entity)
        {
            throw new NotImplementedException();
        }
    }
}
